main.js

```diff
@@ -3230,18 +3229,18 @@ async function startInstance(id, wakeUp) {
         args.push(`--max-old-space-size=${parseInt(instance.common.memoryLimitMB, 10)}`);
     }
 
-    let fileNameFull = path.join(adapterDir, fileName);
-
     // workaround for old vis.
     if (instance.common.onlyWWW && name === 'vis') {
         instance.common.onlyWWW = false;
     }
 
-    if (instance.common.mode !== 'extension' && (instance.common.onlyWWW || !fs.existsSync(fileNameFull))) {
-        fileName = name + '.js';
-        fileNameFull = path.join(adapterDir, fileName);
-        if (instance.common.onlyWWW || !fs.existsSync(fileNameFull)) {
-            // If not just www files
+    let adapterMainFile;
+    try {
+        adapterMainFile = await tools.resolveAdapterMainFile(name);
+    } catch {
+        // The main file was not found or doesn't exist. Doesn't matter for extensions.
+        // TODO: explain why it does not matter!
```

Comment:
afaik extensions only provide html files, icons, widgets etc., that's why it does not matter

Reply:
in fact they should not have a main file defined if not needed :-) but yes theory and reality. We could verify that

Reply:
Ok, the 4 extensions we have right now all have no main defined at all

Reply:
@AlCalzone I somehoe miss here the "also execute that code if file is defined and found but onlyWWW defined ... should never happen ... but